### PR TITLE
[17.07.x] Changelog

### PR DESCRIPTION
**Pull request by vieux**
ping @andrewhsu anything to add packaging wise ?

ping @thaJeztah

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Password stdin [docker/cli#271](https://github.com/docker/cli/pull/271)
+* Don't attempt to remove unsupported resources on older daemon [docker/cli#277](https://github.com/docker/cli/pull/277)
+* Return an empty stats if "container not found" [moby/moby#34004](https://github.com/moby/moby/pull/34004)
++ Add cluster events details in swagger.yml [moby/moby#34035](https://github.com/moby/moby/pull/34035)
+* Matching events using ID and name [moby/moby#33377](https://github.com/moby/moby/pull/33377)
+- Fix stack output stream [docker/cli#298](https://github.com/docker/cli/pull/298)
+* Progress: Show task error in place of progress bar [docker/cli#259](https://github.com/docker/cli/pull/259)
+* Login: print a big warning when using --password [docker/cli#270](https://github.com/docker/cli/pull/270)
++ Add `--detach` to docker scale [docker/cli#243](https://github.com/docker/cli/pull/243)
+* Set ping version even on error [moby/moby#33827](https://github.com/moby/moby/pull/33827)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Make API version negotiation more robust

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Progress: Show success message when finished counting down [docker/cli#260](https://github.com/docker/cli/pull/260)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Perhaps we should combine the "Progress" bullets to a single "improve presentation of progress bars in interactive mode"

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Hide `--detach` for docker < 17.05 [docker/cli#219](https://github.com/docker/cli/pull/219)
+* Progress: Show progress of replicated tasks before they are assigned [docker/cli#237](https://github.com/docker/cli/pull/237)
+* Improve volume spec parsing [docker/cli#152](https://github.com/docker/cli/pull/152)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Not really a user facing change - guess we can either skip it, or have an "other" section

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Allow Proxy Configuration in config.json [docker/cli#93](https://github.com/docker/cli/pull/93)
```
Comment (thaJeztah) [inline CHANGELOG.md]: hm, looks like the docs weren't updated for this one 😞 https://github.com/docker/cli/pull/93#issuecomment-317113658

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
++ Add `scope` filter in `GET /networks/(id or name)` [moby/moby#33630](https://github.com/moby/moby/pull/33630)
```
Comment (thaJeztah) [inline CHANGELOG.md]: This is the API side; perhaps we should put the API in its own section, mention which API version we bumped to, and copy the API changes from https://github.com/moby/moby/blob/master/docs/api/version-history.md (or refer to it)

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Enable client side digest pinning for stack deploy [docker/cli#121](https://github.com/docker/cli/pull/121)
+* Allow --detach and --quiet flags when using --rollback [docker/cli#144](https://github.com/docker/cli/pull/144)
```
Comment (thaJeztah) [inline CHANGELOG.md]: This was cherry picked into 17.06

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Login: print a big warning when using --password [docker/cli#270](https://github.com/docker/cli/pull/270)
++ Add `--detach` to docker scale [docker/cli#243](https://github.com/docker/cli/pull/243)
+* Set ping version even on error [moby/moby#33827](https://github.com/moby/moby/pull/33827)
+* Progress: Show success message when finished counting down [docker/cli#260](https://github.com/docker/cli/pull/260)
+* Hide `--detach` for docker < 17.05 [docker/cli#219](https://github.com/docker/cli/pull/219)
+* Progress: Show progress of replicated tasks before they are assigned [docker/cli#237](https://github.com/docker/cli/pull/237)
+* Improve volume spec parsing [docker/cli#152](https://github.com/docker/cli/pull/152)
+* Allow Proxy Configuration in config.json [docker/cli#93](https://github.com/docker/cli/pull/93)
++ Add `scope` filter in `GET /networks/(id or name)` [moby/moby#33630](https://github.com/moby/moby/pull/33630)
+* Enable client side digest pinning for stack deploy [docker/cli#121](https://github.com/docker/cli/pull/121)
```
Comment (thaJeztah) [inline CHANGELOG.md]: This was cherry picked into 17.06

File: CHANGELOG.md [hunk 8]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Allow --detach and --quiet flags when using --rollback [docker/cli#144](https://github.com/docker/cli/pull/144)
+
+### Builder
+
++ Add DOCKER_BUILD_ARGS to support build with proxy [docker/cli#335](https://github.com/docker/cli/pull/335)
```
Comment (thaJeztah) [inline CHANGELOG.md]: This is a change to the Makefile, so can be removed from the changelog

File: CHANGELOG.md [hunk 9]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+- Fix multiple `COPY --from` in multistage builds [moby/moby#34063](https://github.com/moby/moby/pull/34063)
```
Comment (thaJeztah) [inline CHANGELOG.md]: This was a fix for master, so can be removed

File: CHANGELOG.md [hunk 10]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
++ Add path filtering to build session client [moby/moby#33859](https://github.com/moby/moby/pull/33859)
```
Comment (thaJeztah) [inline CHANGELOG.md]: This, together with "implement long running interactive session", and "incrementally send build context" is all the same feature; perhaps @tonistiigi has a good description for the changelog

File: CHANGELOG.md [hunk 11]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+- Fix handling of remote "git@" notation [moby/moby#33696](https://github.com/moby/moby/pull/33696)
```
Comment (thaJeztah) [inline CHANGELOG.md]: This is marked for cherry-picking into 17.06

File: CHANGELOG.md [hunk 12]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Incrementally sending build context [docker/cli#231](https://github.com/docker/cli/pull/231)
+* Warn on empty continuation lines [moby/moby#33719](https://github.com/moby/moby/pull/33719)
+* Implement long running interactive session and sending build context incrementally [moby/moby#32677](https://github.com/moby/moby/pull/32677)
+- Fix copy —from conflict with force pull [moby/moby#33735](https://github.com/moby/moby/pull/33735)
```
Comment (thaJeztah) [inline CHANGELOG.md]: This is marked for cherry picking into 17.06 https://github.com/docker/docker-ce/issues/86

ping @tiborvass ^^

File: CHANGELOG.md [hunk 13]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Move file coping from the daemon to the builder [moby/moby#33454](https://github.com/moby/moby/pull/33454)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Internal refactoring, think we can skip this one

File: CHANGELOG.md [hunk 14]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+- Fix behavior of absolute paths in .dockerignore [moby/moby#32088](https://github.com/moby/moby/pull/32088)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Fixed a bug where .dockerignore file entries with a leading `/` would not match anything

File: CHANGELOG.md [hunk 15]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+- Fix Cache with ONBUILD [moby/moby#33524](https://github.com/moby/moby/pull/33524) [moby/moby#33500](https://github.com/moby/moby/pull/33500)
```
Comment (thaJeztah) [inline CHANGELOG.md]: These are already cherry picked into 17.06

File: CHANGELOG.md [hunk 16]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Refactor builder probe cache and container backend [moby/moby#33061](https://github.com/moby/moby/pull/33061)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Internal refactoring, may not be worth mentioning

File: CHANGELOG.md [hunk 17]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+
+### Distribution
+
+* Stop trying to load images on an incompatible OS [moby/moby#33647](https://github.com/moby/moby/pull/33647)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Improve error message when attempting to load an image on an incompatible OS.

Comment (thaJeztah) [inline CHANGELOG.md]: Actually, this is backported to 17.06 already

File: CHANGELOG.md [hunk 18]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Do not add duplicate platform information to service spec [moby/moby#33867](https://github.com/moby/moby/pull/33867)
```
Comment (thaJeztah) [inline CHANGELOG.md]: This is already backported to 17.06

File: CHANGELOG.md [hunk 19]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+
+### Logging
+
+- Fix log readers can block writes indefinitely [moby/moby#33897](https://github.com/moby/moby/pull/33897)
```
Comment (thaJeztah) [inline CHANGELOG.md]: This is backported to 17.06

File: CHANGELOG.md [hunk 20]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
++ Add stream format details for attach/logs endpoint [moby/moby#33494](https://github.com/moby/moby/pull/33494)
```
Comment (thaJeztah) [inline CHANGELOG.md]: This is a GoDoc only change, no code changes, so can be removed

File: CHANGELOG.md [hunk 21]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+- Fix wrong filemode for rotate log files [moby/moby#33926](https://github.com/moby/moby/pull/33926)
+* Expose config credentials without needing the Cli [docker/cli#227](https://github.com/docker/cli/pull/227)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Internal refactoring, can be removed

File: CHANGELOG.md [hunk 22]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* System prune: only warn for volumes if --volumes is given [docker/cli#255](https://github.com/docker/cli/pull/255)
```
Comment (thaJeztah) [inline CHANGELOG.md]: This was backported to 17.06, so can be removed

File: CHANGELOG.md [hunk 23]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+- Fix stderr logging for journald and syslog [moby/moby#33832](https://github.com/moby/moby/pull/33832)
+* Pass the same log filename to logging plugins on start/stop requests [moby/moby#33583](https://github.com/moby/moby/pull/33583)
```
Comment (thaJeztah) [inline CHANGELOG.md]: This was backported to 17.06 so can be removed

File: CHANGELOG.md [hunk 24]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+
+### Networking
+
+* Include Endpoint List for Shared Endpoints [moby/moby#33662](https://github.com/moby/moby/pull/33662)
+* LCOW: owner and network endpoints [moby/moby#33774](https://github.com/moby/moby/pull/33774)
```
Comment (thaJeztah) [inline CHANGELOG.md]: This (possibly the one above too) look like internal changes

File: CHANGELOG.md [hunk 25]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Host and Bridge network support in docker stack deploy [docker/cli#132](https://github.com/docker/cli/pull/132)
```
Comment (thaJeztah) [inline CHANGELOG.md]: This is in 17.06 already

File: CHANGELOG.md [hunk 26]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+
+### Runtime
+
+* Keep pause state when restoring container's status [moby/moby#34071](https://github.com/moby/moby/pull/34071)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Fixes an issue on master, not in a release IIUC

File: CHANGELOG.md [hunk 27]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Plugable secret backend [moby/moby#34157](https://github.com/moby/moby/pull/34157) [moby/moby#34123](https://github.com/moby/moby/pull/34123)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Swarm mode; this is a feature - use external secret stores (!)

- CLI is not merged yet btw

File: CHANGELOG.md [hunk 28]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Small refactor in system prune [docker/cli#341](https://github.com/docker/cli/pull/341)
```
Comment (thaJeztah) [inline CHANGELOG.md]: No user facing changes, can be removed

File: CHANGELOG.md [hunk 29]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Store container names in memdb [moby/moby#33886](https://github.com/moby/moby/pull/33886)
+* In the case of remounting with changed data, need to call mount [moby/moby#34077](https://github.com/moby/moby/pull/34077)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Will be cherry picked into 17.06

(description could be "Fix secrets permission errors with `userns-remap` enabled")

File: CHANGELOG.md [hunk 30]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* LCOW: Service VM lifetime changes [moby/moby#33969](https://github.com/moby/moby/pull/33969)
```
Comment (thaJeztah) [inline CHANGELOG.md]: LCOW is all WIP, so can probably be removed

File: CHANGELOG.md [hunk 31]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Enable inspect exec if container is pause/restarting/non-running [moby/moby#32881](https://github.com/moby/moby/pull/32881)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Fix race condition between `docker exec` and `docker pause`

File: CHANGELOG.md [hunk 32]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Allow stopping of paused container [moby/moby#34027](https://github.com/moby/moby/pull/34027)
```
Comment (thaJeztah) [inline CHANGELOG.md]: This is an enhancement, could be moved more "up"

File: CHANGELOG.md [hunk 33]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Devicemapper: rework logging and add --storage-opt dm.libdm_log_level [moby/moby#33845](https://github.com/moby/moby/pull/33845)
+* Devicemapper: Wait for device removal if deferredRemoval=true and deferredDeletion=… [moby/moby#33877](https://github.com/moby/moby/pull/33877)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Devicemapper: Prevent "device in use" errors if deferred removal is enabled, but not deferred deletion

File: CHANGELOG.md [hunk 34]
```diff
@@ -5,6 +5,121 @@ information on the list of deprecated flags and APIs please have a look at
...
+* Return prune data when context canceled [moby/moby#33979](https://github.com/moby/moby/pull/33979)
```
Comment (thaJeztah) [inline CHANGELOG.md]: Note to self: here's where I left off

**Comment by rogaha** [comment]
@vieux we need to add the `telemetry plugin`, it will be installed, but `disabled` by default.

**Comment by thaJeztah** [comment]
@vieux for some reason it didn't push to your branch, but you can take this commit; https://github.com/thaJeztah/docker-ce/commit/19f70adcf6d907a8919f1d189846581e31b67b33

**Comment by andrewhsu** [comment]
LGTM

File: CHANGELOG.md [hunk 35]
```diff
@@ -5,6 +5,68 @@ information on the list of deprecated flags and APIs please have a look at
...
+
+* Allow stopping of paused container [moby/moby#34027](https://github.com/moby/moby/pull/34027)
++ Add quota support for the overlay2 storage driver [moby/moby#32977](https://github.com/moby/moby/pull/32977)
+* Remove container locks on `docker ps` [moby/moby#31273](https://github.com/moby/moby/pull/31273)
+* Store container names in memdb [moby/moby#33886](https://github.com/moby/moby/pull/33886)
+* Fix race condition between `docker exec` and `docker pause` [moby/moby#32881](https://github.com/moby/moby/pull/32881)
+* Devicemapper: Rework logging and add `--storage-opt dm.libdm_log_level` [moby/moby#33845](https://github.com/moby/moby/pull/33845)
+* Devicemapper: Prevent "device in use" errors if deferred removal is enabled, but not deferred deletion [moby/moby#33877](https://github.com/moby/moby/pull/33877)
+* Devicemapper: Use KeepAlive to prevent tasks being garbage-collected while still in use [moby/moby#33376](https://github.com/moby/moby/pull/33376)
+* Report inetermediate prune results if prune is cancelled [moby/moby#33979](https://github.com/moby/moby/pull/33979)
```
Comment (andyneff) [inline CHANGELOG.md]: Spelling: inetermediate -> intermediate

**Pull request merged**